src/creature.cpp

```diff
@@ -102,6 +103,24 @@ void Creature::reset()
     reset_bonuses();
     reset_stats();
 }
+
+void Creature::bleed( const tripoint &where, int intensity ) const
```

Comment:
> There are even more of them scattered over the code. IMHO there should me as few of such functions, as possible, and they shouldn't belong neither to ranged.cpp nor vehicle.cpp.

Could be wrapped up as `map::splatter` taking blood type, point of origin and endpoint, handling things like blood trail stopping on an obstacle.
Alternatively it could take a list of points to splatter blood on, but that would leave trajectory logic to caller, not reducing duplication all that much.